### ISSUE TITLE
Flatten field

### ISSUE THREAD

**Issue by fulmicoton** [quickwit-inc/quickwit#310]
It is not always a possibility to have a strict mapping.
In that case we want to be able to map fields to a dynamic field, and still enjoyed full text search.

One approach is to encode tokens in a special way to encode the structure of the object.
For instance

```
{
   "text": "my text",
   "title: "hello",
   "author": {
      "name": "Roger" 
   }
}
```

Could be emit the tokens in a dynamic field, with the following as the token values:
```
text:my
text:text
title:hello
author.name:Roger
```

**Comment by fmassot** [comment]
I would name this differently. I understand dynamic mapping as a mapping which will change through time according to changes in fields in indexed documents. 
What you describe is more like the flatten field in elastic: https://www.elastic.co/guide/en/elasticsearch/reference/master/flattened.html

In fact, my idea was to do that in tantivy, I opened the discussion here: https://github.com/tantivy-search/tantivy/issues/1050

**Comment by fulmicoton** [comment]
You are right. I edited the title.

**Comment by fulmicoton** [comment]
Closed in favor for #1022
